chatsky/pipeline/pipeline/pipeline.py

```diff
@@ -83,61 +82,62 @@ class Pipeline:
 
     """
 
-    def __init__(
-        self,
-        components: ServiceGroupBuilder,
-        script: Union[Script, Dict],
-        start_label: NodeLabel2Type,
-        fallback_label: Optional[NodeLabel2Type] = None,
-        label_priority: float = 1.0,
-        condition_handler: Optional[Callable] = None,
-        slots: Optional[Union[GroupSlot, Dict]] = None,
-        handlers: Optional[Dict[ActorStage, List[Callable]]] = None,
-        messenger_interface: Optional[MessengerInterface] = None,
-        context_storage: Optional[Union[DBContextStorage, Dict]] = None,
-        before_handler: Optional[ExtraHandlerBuilder] = None,
-        after_handler: Optional[ExtraHandlerBuilder] = None,
-        timeout: Optional[float] = None,
-        optimization_warnings: bool = False,
-        parallelize_processing: bool = False,
-    ):
-        self.actor: Actor = None
-        self.messenger_interface = CLIMessengerInterface() if messenger_interface is None else messenger_interface
-        self.context_storage = {} if context_storage is None else context_storage
-        self.slots = GroupSlot.model_validate(slots) if slots is not None else None
-        self._services_pipeline = ServiceGroup(
-            components,
-            before_handler=before_handler,
-            after_handler=after_handler,
-            timeout=timeout,
+    pre_services: ServiceGroup = Field(default_factory=list)
+    post_services: ServiceGroup = Field(default_factory=list)
+    script: Union[Script, Dict]
+    start_label: NodeLabel2Type
+    fallback_label: Optional[NodeLabel2Type] = None
+    label_priority: float = 1.0
+    condition_handler: Callable = Field(default=default_condition_handler)
+    slots: GroupSlot = Field(default_factory=GroupSlot)
+    handlers: Dict[ActorStage, List[Callable]] = Field(default_factory=dict)
+    messenger_interface: MessengerInterface = Field(default_factory=CLIMessengerInterface)
+    context_storage: Union[DBContextStorage, Dict] = Field(default_factory=dict)
+    before_handler: ComponentExtraHandler = Field(default_factory=list)
+    after_handler: ComponentExtraHandler = Field(default_factory=list)
+    timeout: Optional[float] = None
+    optimization_warnings: bool = False
+    parallelize_processing: bool = False
+    actor: Optional[Actor] = None
+    _services_pipeline: Optional[ServiceGroup] = None
+    _clean_turn_cache: Optional[bool]
+
+    def _create_actor(self) -> Actor:
```

Comment:
I browsed the net, and in case someone directly changes a constructor parameter in Pipeline, `computed_field` will start recomputing, breaking the program. I'm not sure what happens if the changed parameter is unrelated to said `computed field`, like when a Pipeline's flag is changed after it's creation.
In case the change is indirect, like "pre_services.components[0].set_state(FINISHED)" the program probably won't start recomputing `_services_pipeline`, right? I haven't found much on the internet about this.